src/vm/i386/cgenx86.cpp

```diff
@@ -303,6 +305,26 @@ void TransitionFrame::UpdateRegDisplayHelper(const PREGDISPLAY pRD, UINT cbStack
     pRD->ControlPC = *PTR_PCODE(pRD->PCTAddr);
     pRD->Esp  = (DWORD)(pRD->PCTAddr + sizeof(TADDR) + cbStackPop);
 
+#ifdef WIN64EXCEPTIONS
+    pRD->IsCallerContextValid = FALSE;
+    pRD->IsCallerSPValid      = FALSE;
+
+    pRD->pCurrentContext->Eip = pRD->ControlPC;
+    pRD->pCurrentContext->Esp = GetSP();
```

Comment:
A have looked into how this should work in detail. The difference between the value returned by GetSP and the value set to pRD->Esp is that the pRD->Esp = GetSP() + cbStackPop. In other words, one represent the SP before pushing the stack arguments at the caller and the other SP after pushing them. 
So the pRD->Esp should be set for WIN64EXCEPTIONS to the GetSP too. the pEsp and the pRD->pCurrentContext->Esp need to match.
And in fact, it happens later in the SyncRegDisplayToCurrentContext. So the lines 305 and 306 are not needed for WIN64EXCEPTIONS if you change the line 312 to
```C++
pRD->pCurrentContext->Eip = *PTR_PCODE(pRD->PCTAddr);
```
Could you please make this change and change the lines 305 and 306 to be for !WIN64EXCEPTIONS only?